sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/kllAggregates.scala

```diff
@@ -0,0 +1,405 @@
+    this.createAggregationBuffer()
+  }
+}
+
+/**
+ * The KllSketchAggFloat function utilizes an Apache DataSketches KllFloatsSketch instance to
+ * compute quantiles of the values of an input expression (such as an input column in a table).
+ * It outputs the binary representation of the KllFloatsSketch.
+ *
+ * See [[https://datasketches.apache.org/docs/KLL/KLLSketch.html]] for more information.
+ *
+ * @param child
+ *   child expression against which quantile computation will occur
+ * @param mutableAggBufferOffset
+ *   offset for mutable aggregation buffer
+ * @param inputAggBufferOffset
+ *   offset for input aggregation buffer
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_(expr) - Returns the KllFloatsSketch compact binary representation.
+  """,
+  examples = """
+    Examples:
+      > SELECT LENGTH(kll_sketch_to_string_float(_FUNC_(col))) > 0 FROM VALUES (1), (2), (3), (4), (5) tab(col);
+       true
+  """,
+  group = "agg_funcs",
+  since = "4.1.0")
+// scalastyle:on line.size.limit
+case class KllSketchAggFloat(
+    override val child: Expression,
+    override val mutableAggBufferOffset: Int = 0,
+    override val inputAggBufferOffset: Int = 0)
+    extends TypedImperativeAggregate[KllBufferFloat]
+        with UnaryLike[Expression]
+        with ExpectsInputTypes {
+  def this(child: Expression) = this(child, 0, 0)
+
+  override def withNewMutableAggBufferOffset(newMutableAggBufferOffset: Int): KllSketchAggFloat =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): KllSketchAggFloat =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+  override protected def withNewChildInternal(newInput: Expression): KllSketchAggFloat =
+    copy(child = newInput)
+
+  override def dataType: DataType = BinaryType
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(
+      TypeCollection(
+        ByteType,
+        FloatType,
+        IntegerType,
+        LongType,
+        ShortType
+      ))
+  override def nullable: Boolean = false
+  override def prettyName: String = "kll_sketch_agg_float"
+  override def createAggregationBuffer(): KllBufferFloat = KllBufferFloat()
+
+  /**
+   * Evaluate the input row and update the KllFloatsSketch instance with the row's value. The update
+   * function only supports a subset of Spark SQL types, and an exception will be thrown for
+   * unsupported types.
+   * Note, Null values are ignored.
+   */
+  override def update(updateBuffer: KllBufferFloat, input: InternalRow): KllBufferFloat = {
+    // Return early for null values.
+    val v = child.eval(input)
+    if (v == null) {
+      return updateBuffer
+    }
+    // Handle the different data types for sketch updates.
+    val sketch: KllFloatsSketch = updateBuffer.sketch
+    child.dataType match {
+      case ByteType =>
+        sketch.update(v.asInstanceOf[Byte].toFloat)
+      case IntegerType =>
```

Comment:
I believe this integer to float conversion may also result in loss of precision in some cases.

Reply:
Sure, this is done.